### PR TITLE
Feature/85 allow users to change their password via profile

### PR DESCRIPTION
**Pull request by pro100proger**

### CODE REVIEW

**Review by ostartt — APPROVED** [review]

File: src/main/java/com/softserve/edu/sporthubujp/controller/UserController.java [hunk 1]
```diff
@@ -38,4 +42,14 @@ public ResponseEntity<UserDTO> updateProfile(@NotNull Principal principal,
         return ResponseEntity.status(HttpStatus.OK).body(
             userService.updateUser(user, newUser));
     }
+
+    @PutMapping(path = "/change/password")
```
Comment (yuremboo) [inline src/main/java/com/softserve/edu/sporthubujp/controller/UserController.java:46]: /password

File: src/main/resources/application.properties [hunk 2]
```diff
@@ -1,9 +1,8 @@
 spring.datasource.username=postgres
-spring.datasource.password=this1pass
-
+spring.datasource.password=Lv_2022_06
 spring.datasource.url=jdbc:postgresql://localhost:5432/sporthubujp
 spring.jpa.database-platform=org.hibernate.dialect.PostgreSQLDialect
-spring.jpa.hibernate.ddl-auto = validate
+spring.jpa.hibernate.ddl-auto = update
```
Comment (yuremboo) [inline src/main/resources/application.properties:5]: validate

File: src/test/java/com/softserve/edu/sporthubujp/service/impl/UserServiceImplTest.java [hunk 3]
```diff
@@ -113,4 +113,12 @@ void canEnableUser() {
     void canFindUserByEmail() {
 
     }
+
+    @Test
+    void updateUser() {
```
Comment (yuremboo) [inline src/test/java/com/softserve/edu/sporthubujp/service/impl/UserServiceImplTest.java:118]: nice tests :)))))

**Review by yuremboo — CHANGES_REQUESTED** [review]

File: src/test/java/com/softserve/edu/sporthubujp/service/impl/UserServiceImplTest.java (hunk 3 above)
Comment (pro100proger) [reply src/test/java/com/softserve/edu/sporthubujp/service/impl/UserServiceImplTest.java:118]: It's preparation before testing)

**Review by NastiaChaikovska — APPROVED** [review]

File: src/main/java/com/softserve/edu/sporthubujp/controller/ArticleController.java [hunk 4]
```diff
@@ -58,7 +58,7 @@ public ArticleController(ArticleService articleService, CommentService commentSe
     @PreAuthorize("hasAnyAuthority('USER', 'ADMIN')")
     public ResponseEntity<ArticleDTO> getArticleById(@PathVariable String id) {
         log.info("Get article by id {}", id);
-        CompletableFuture.supplyAsync(() -> logRepository.save(new Logs(id)));
+//        CompletableFuture.supplyAsync(() -> logRepository.save(new Logs(id)));
```
Comment (yuremboo) [inline src/main/java/com/softserve/edu/sporthubujp/controller/ArticleController.java:61]: do not comment code

File: src/main/java/com/softserve/edu/sporthubujp/service/UserService.java [hunk 5]
```diff
@@ -21,7 +22,10 @@ public interface UserService {
 
     UserDTO updateUser(User oldUser, UserSaveProfileDTO newUser);
 
+    UserDTO updatePassword(User oldUser, UserSavePasswordDTO newUser) throws InvalidPropertiesFormatException;
```
Comment (yuremboo) [inline src/main/java/com/softserve/edu/sporthubujp/service/UserService.java:25]: this is not oldUser :)

File: src/main/java/com/softserve/edu/sporthubujp/service/impl/UserServiceImpl.java [hunk 6]
```diff
@@ -130,33 +134,51 @@ public UserDTO updateUser(User oldUser, UserSaveProfileDTO newUser) {
             userMapper.updateUser(oldUser, newUser))
         );
     }
+    public UserDTO updatePassword(User oldUser, UserSavePasswordDTO passwords) throws InvalidPropertiesFormatException {
+        boolean checkPasswords = passwordConfig.passwordEncoder().matches(passwords.getOldPassword(), oldUser.getPassword());
+        if (checkPasswords) {
+            if (passwordValidator.test(passwords.getPassword())) {
+                passwords.setPassword(passwordConfig.passwordEncoder().encode(passwords.getPassword()));
+            } else {
+                throw new InvalidPropertiesFormatException("Service: password must contain at least 8 characters (letters and numbers)");
```
Comment (yuremboo) [inline src/main/java/com/softserve/edu/sporthubujp/service/impl/UserServiceImpl.java:143]: Use application based exception, like ServiceException etc.

**Review by yuremboo — CHANGES_REQUESTED** [review]

File: src/main/java/com/softserve/edu/sporthubujp/service/impl/UserServiceImpl.java [hunk 7]
```diff
@@ -105,10 +110,10 @@ public UserSaveProfileDTO findUserById(String userId) {
         return userMapper.userToUserSaveDto(userRepository.findUserById(userId));
     }
 
-//    @Override
-//    public User findUserByPasswordResetToken(String token) {
-//        return null;
-//    }
+    //    @Override
```
Comment (yuremboo) [inline src/main/java/com/softserve/edu/sporthubujp/service/impl/UserServiceImpl.java:113]: remove commented code

**Review by yuremboo — APPROVED** [review]

**Pull request merged**